src/inet/DNSResolver.cpp

```diff
@@ -91,9 +91,9 @@ INET_ERROR DNSResolver::Resolve(const char *hostName, uint16_t hostNameLen, uint
         DNSResolver::OnResolveCompleteFunct onComplete, void *appState)
 {
     INET_ERROR res = INET_NO_ERROR;
+    Weave::System::Object::AutoRelease objectRelease(*this);
```

Comment:
Similar to my comment below, I don't see this as a desirable pattern.  Rather, I would argue that the returns should be replaced with ExitNow() jumps to an exit: label containing a single call to Release().